### PR TITLE
update integration test for tensorflow model load/save on hdfs/s3

### PR DESCRIPTION
**Pull request by frankfzw**
## What changes were proposed in this pull request?

1. Fix data format mismatch in integration test of tensorflow model load/save on hdfs and s3 
2. Change data format in `TensorflowSaverSpec` to `NHWC` except `BatchNormalization` since BigDL has not support `NHWC` `BatchNormalization` yet.
3. Remove format transformation while saving BigDL model to Tensorflow model. The saved tf model will have same format as the original BigDL model.
4. Add Reshape node in `BatchNorm2DToTF` to support `NCHW` batch normalization. The dimension size 
 of ReShape is set to [1, C, 1, 1] since the `BatchNormalization` in BigDL is `NCHW`. The new topology of saved graph is same as `tf.layers.batch_normalization`.
graph-run

## How was this patch tested?

unit tests, integration tests

### CODE REVIEW

**Comment by yiheng** [comment]
@yangw1234 please take a look at this PR. Looks like your previous change https://github.com/intel-analytics/BigDL/pull/1506  break the integration test.

**Comment by yangw1234** [comment]
@frankfzw Hi, can you also change the tensorflow saver code, so that a bigdl NCHW model saves to a tensorflow NCHW model and so as NHWC. Then we can get rid of the `transpose` code in this test case, which is very confusing. cc @yiheng

**Comment by frankfzw** [comment]
@yangw1234 I can add data format in `TensorflowSaver` and refactor the corresponding test code. cc @yiheng

**Comment by frankfzw** [comment]
@yangw1234 The unit tests are all passed under tensorflow cpu, tensorflow mkl and tensorflow gpu in a local server. cc @yiheng

**Comment by frankfzw** [comment]
pr validation and integration tests passed

**Comment by yangw1234** [comment]
LGTM
